File: server/storage.ts
import type { AnalyzeLogsResponse, AnalysisHistoryItem } from "@shared/schema";
import { randomUUID } from "crypto";

export interface IStorage {
  saveAnalysis(analysis: AnalyzeLogsResponse): Promise<AnalyzeLogsResponse>;
  getAnalysis(id: string): Promise<AnalyzeLogsResponse | undefined>;
  getAnalysisHistory(): Promise<AnalysisHistoryItem[]>;
}

export class MemStorage implements IStorage {
  private analyses: Map<string, AnalyzeLogsResponse>;

  constructor() {
    this.analyses = new Map();
  }

  async saveAnalysis(analysis: AnalyzeLogsResponse): Promise<AnalyzeLogsResponse> {
    this.analyses.set(analysis.id, analysis);
    return analysis;
  }

  async getAnalysis(id: string): Promise<AnalyzeLogsResponse | undefined> {
    return this.analyses.get(id);
  }

  async getAnalysisHistory(): Promise<AnalysisHistoryItem[]> {
    return Array.from(this.analyses.values())
      .map((a) => ({
        id: a.id,
        createdAt: a.createdAt,
        logCount: a.parsedLogs.length,
        errorCount: a.errorGroups.reduce((sum, g) => sum + g.count, 0),
        status: a.status,
      }))
      .sort((a, b) => new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime());
  }
}

export const storage = new MemStorage();

File: api/storage.py
from typing import Dict, List, Optional
from api.schemas import AnalyzeLogsResponse, AnalysisHistoryItem

class MemStorage:
    def __init__(self):
        self.analyses: Dict[str, AnalyzeLogsResponse] = {}
    
    async def save_analysis(self, analysis: AnalyzeLogsResponse) -> AnalyzeLogsResponse:
        self.analyses[analysis.id] = analysis
        return analysis
    
    async def get_analysis(self, id: str) -> Optional[AnalyzeLogsResponse]:
        return self.analyses.get(id)
    
    async def get_analysis_history(self) -> List[AnalysisHistoryItem]:
        items = []
        for a in self.analyses.values():
            items.append(AnalysisHistoryItem(
                id=a.id,
                createdAt=a.createdAt,
                logCount=len(a.parsedLogs),
                errorCount=sum(g.count for g in a.errorGroups),
                status=a.status
            ))
        items.sort(key=lambda x: x.createdAt, reverse=True)
        return items

storage = MemStorage()
